client/src/components/TreeImageScrubber.js

```diff
@@ -142,24 +156,22 @@ function TreeImageScrubber({ classes, getScrollContainerRef, ...props }) {
     loadMoreTreeImages();
   };
 
+  scrollContainerRef = getScrollContainerRef() ? getScrollContainerRef() : null;
+  if (scrollContainerRef) {
+    scrollContainerRef.addEventListener("scroll", handleScroll);
+  }
+
   useEffect(() => {
-    scrollContainerRef = getScrollContainerRef()
-      ? getScrollContainerRef()
-      : null;
-    if (treeImages.length === 0) {
-      console.log("at 0 again!", state);
+    if (needtoLoadMoreTreeImages()) {
       loadMoreTreeImages();
     }
 
-    if (scrollContainerRef) {
-      scrollContainerRef.addEventListener("scroll", handleScroll);
-    }
     return () => {
       if (scrollContainerRef) {
         scrollContainerRef.removeEventListener("scroll", handleScroll);
       }
     };
-  }, []);
+  }, [state]);
```

Comment:
possibly weird, but more specific state change checks seems to make things slower.. we'll see

Reply:
why is this possibly weird?

Reply:
It's not weird on second thought. In the context of using redux, it might be a bad idea to have one component check state that is relevant to all component state. but here, it seems just right